services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -1790,6 +1791,87 @@ class HakemusServiceITest(
+            val expectedJohtoselvityshakemusDataAfterSend =
+                (applicationEntity.hakemusEntityData as KaivuilmoitusEntityData)
+                    .createAccompanyingJohtoselvityshakemusData()
+                    .toHakemusData(hakemusData.yhteystiedotMap())
+                    .setOrdererForContractor(founder.id)
+            val expectedKaivuilmoitusDataAfterSend =
+                hakemusData.copy(pendingOnClient = false).setOrdererForContractor(founder.id)
+            val expectedCableReportAlluRequest =
+                expectedJohtoselvityshakemusDataAfterSend.toAlluCableReportData(hakemus.hankeTunnus)
+            val expectedExcavationAnnouncementAlluRequest =
+                expectedKaivuilmoitusDataAfterSend.toAlluExcavationNotificationData(
+                    hakemus.hankeTunnus)
+            val cableReportAlluId = alluId
+            val excavationAnnouncenemtAlluId = alluId + 1
+            every { alluClient.create(expectedCableReportAlluRequest) } returns cableReportAlluId
+            every { alluClient.create(expectedExcavationAnnouncementAlluRequest) } returns
+                excavationAnnouncenemtAlluId
+            justRun { alluClient.addAttachment(cableReportAlluId, any()) }
+            justRun { alluClient.addAttachment(excavationAnnouncenemtAlluId, any()) }
+            justRun { alluClient.addAttachments(cableReportAlluId, any(), any()) }
+            justRun { alluClient.addAttachments(excavationAnnouncenemtAlluId, any(), any()) }
+            every { alluClient.getApplicationInformation(cableReportAlluId) } returns
+                AlluFactory.createAlluApplicationResponse(cableReportAlluId)
+            every { alluClient.getApplicationInformation(excavationAnnouncenemtAlluId) } returns
+                AlluFactory.createAlluApplicationResponse(
+                    excavationAnnouncenemtAlluId,
+                    applicationId = DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+
+            val response = hakemusService.sendHakemus(hakemus.id, USERNAME)
+
+            assertThat(response).all {
+                prop(Hakemus::alluid).isEqualTo(excavationAnnouncenemtAlluId)
+                prop(Hakemus::applicationIdentifier)
+                    .isEqualTo(ApplicationFactory.DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+                prop(Hakemus::alluStatus).isEqualTo(ApplicationStatus.PENDING)
+                prop(Hakemus::applicationData).isEqualTo(expectedKaivuilmoitusDataAfterSend)
+            }
+            assertThat(hakemusRepository.getReferenceById(hakemus.id)).all {
+                prop(HakemusEntity::alluid).isEqualTo(excavationAnnouncenemtAlluId)
+                prop(HakemusEntity::applicationIdentifier)
+                    .isEqualTo(ApplicationFactory.DEFAULT_EXCAVATION_ANNOUNCEMENT_IDENTIFIER)
+                prop(HakemusEntity::alluStatus).isEqualTo(ApplicationStatus.PENDING)
+            }
+            assertThat(hakemusRepository.getReferenceById(hakemus.id + 1)).all {
```

Comment:
Guessing IDs is dangerous. There's a method for finding the hakemus by alluId, which is known.
```suggestion
            assertThat(hakemusRepository.getOneByAlluid(cableReportAlluId)).isNotNull().all {
```

Reply:
You're right. Thanks for pointing out this existing finder function!